### ISSUE TITLE
Parallelize manual page rendering?

### ISSUE THREAD

**Issue by etiennebacher** [etiennebacher/altdoc#82]
Rendering 60+ man pages can be very slow since all man pages have to run their own quarto process. This sort of task sounds ideal for parallelization (would probably require a dependency in Suggests)

Edit: could be implemented in quarto in the future but for now we need to do it ourselves: https://github.com/quarto-dev/quarto-cli/issues/6116#issuecomment-1623392731

**Comment by vincentarelbundock** [comment]
Agreed.

I also want to implement a freeze argument which checksums Rd files, but I don't have a clear view of code design yet.

This will be a separate PR.

**Comment by etiennebacher** [comment]
> I also want to implement a freeze argument which checksums Rd files

That's a bit risky because examples could fail due to changes in source code but we wouldn't detect them since the content didn't actually change. 

This feature can be useful for other parts of the docs development, e.g for styling, but we should accompany it with a big warning

**Comment by vincentarelbundock** [comment]
Yes, definitely not a default.

**Comment by etiennebacher** [comment]
> Unit tests FTW

Sure ;) but also nothing more annoying for users when copy-pasted examples don't work

**Comment by etiennebacher** [comment]
Man page rendering is annoying in `datawizard`, it takes 3min+.

One potential solution, use some arbitrary threshold for "long man pages" and use `future.apply` to render in parallel (tested on a subset of man pages here):

```r
library(future.apply)
plan(multisession)

fs::dir_tree("foo")
```
    foo
    ├── categorize.md
    ├── categorize.qmd
    ├── center.md
    ├── center.qmd
    ├── coerce_to_numeric.md
    ├── coerce_to_numeric.qmd
    ├── convert_na_to.md
    ├── convert_na_to.qmd
    ├── convert_to_na.md
    ├── convert_to_na.qmd
    ├── data_arrange.md
    ├── data_arrange.qmd
    ├── data_codebook.md
    ├── data_codebook.qmd
    ├── data_duplicated.md
    ├── data_duplicated.qmd
    ├── data_extract.md
    ├── data_extract.qmd
    ├── data_group.md
    └── data_group.qmd

``` r
rpts <- list.files("foo", full.names = TRUE, pattern = "\\.qmd$")

system.time({
  future_lapply(rpts, \(x) quarto::quarto_render(input = x, output_format = "md"))
})
```

       user  system elapsed 
       0.47    0.03   10.87 

``` r
fs::file_delete(list.files("foo", full.names = TRUE, pattern = "\\.md"))

system.time({
  for (i in seq_along(rpts)) {
    quarto::quarto_render(input = rpts[i], output_format = "md", quiet = TRUE)
  }
})
```

       user  system elapsed 
       0.26    0.17   28.53 

``` r
fs::file_delete(list.files("foo", full.names = TRUE, pattern = "\\.md"))
```

``` r
pak::pkg_deps_tree("future.apply")
```

    future.apply 1.11.0 [new][dl] (156.59 kB)
    ├─future 1.33.0 [new][dl] (676.25 kB)
    │ ├─digest 0.6.33 [new][dl] (206.21 kB)
    │ ├─globals 0.16.2 [new][dl] (107.01 kB)
    │ │ └─codetools 0.2-19 
    │ ├─listenv 0.9.0 [new][dl] (108.43 kB)
    │ └─parallelly 1.36.0 [new][dl] (351.04 kB)
    └─globals

    Key:  [new] new | [dl] download

**Comment by vincentarelbundock** [comment]
I'm not a fan of switching to parallel automatically with some arbitrary threshold. What if I'm doing something important with the other cores? But again, I think there should be a good `parallel` argument if we have time and energy to implement.

**Comment by vincentarelbundock** [comment]
Perhaps we should make it easy for people to do this, and maybe we should make it default for `altdoc`, but I don't think this should be default for everyone automatically.
